src/PostgREST/Request/Preferences.hs

```diff
@@ -1,54 +1,196 @@
-module PostgREST.Request.Preferences where
+-- |
+-- Module: PostgREST.Request.Preferences
+-- Description: Track client preferences to be employed when processing requests
+--
+-- Track client prefences set in HTTP 'Prefer' headers according to RFC7240[1].
+--
+-- [1] https://datatracker.ietf.org/doc/html/rfc7240
+--
+module PostgREST.Request.Preferences
+  ( Preferences(..)
+  , PreferCount(..)
+  , PreferParameters(..)
+  , PreferRepresentation(..)
+  , PreferResolution(..)
+  , PreferTransaction(..)
+  , fromHeaders
+  , ToAppliedHeader(..)
+  ) where
+
+import qualified Data.ByteString.Char8     as BS
+import qualified Data.Map                  as Map
+import qualified Network.HTTP.Types.Header as HTTP
 
-import GHC.Show
 import Protolude
 
 
+-- $setup
+-- Imports for doctests
+-- >>> import Text.Pretty.Simple (pPrint)
+
+-- | Preferences recognized by the application.
+data Preferences
+  = Preferences
+    { preferResolution     :: Maybe PreferResolution
+    , preferRepresentation :: Maybe PreferRepresentation
+    , preferParameters     :: Maybe PreferParameters
+    , preferCount          :: Maybe PreferCount
+    , preferTransaction    :: Maybe PreferTransaction
+    }
+    deriving Show
```

Comment:
Hm, this line is uncovered and only needed for the doctests. That would be one reason to include the doctests in coverage, otherwise we'd have false reports all over the place again.

Reply:
See above on why we can't get coverage for doctests at this point. At least we can avoid those unneeded derivations in the actual code, see latest commit

Reply:
Yes, I like that!